test/cypress/integration/courses_spec.js

```diff
@@ -0,0 +1,55 @@
+describe("Course", () => {
+
+    describe("As admin", () => {
+
+        let courseUrl;
+
+        beforeEach(() => { 
+            cy.request("/api/testhooks/login_admin");
+            cy.visit("/courses");
+            cy.get('table#course_list')
+                .find('a')
+                .invoke('attr', 'href')
+                .then((href) => {
+                    cy.log(`href=${href}`);
+                    courseUrl = href
+                }).then( () => {
+                    cy.visit(courseUrl)
+                });
+        });
```

Comment:
Are you trying to emulate clicking on a link right here? If so, it might be easier to use [cy.click()](https://docs.cypress.io/api/commands/click.html#Syntax).

Reply:
No, there's more to it than that.   The intent is to find the first courseUrl on the course page, because we need to assert that the "Students" button on the course page links to that specific url.  Just clicking on it won't capture the href value.